recipes/newrelic/apm/java/linux.yml

```diff
@@ -184,434 +123,212 @@ install:
-            echo "CATALINA_OPTS=\"\$CATALINA_OPTS -javaagent:/opt/newrelic/$appName/newrelic.jar\"" >> $configFile
-
-            save_newrelic_configuration $service $user $appName
-
-            touch {{.TMP_DIR}}/tomcat_configured
-          }
-
-          save_sysd_yum_configuration() {
-            local APP_NAME=$1
-            local SERVICE=$1
-            local CONFIG_ROOT=$2
-            local USER=$(ps -q $pid -h -o user)
-
-            if [[ "{{.NEW_RELIC_ASSUME_YES}}" != "true" ]]; then
-              echo -n "Enter an app name (default: $APP_NAME): "
-              read -r answer
-              if [[ -n $answer ]]; then
-                APP_NAME=$answer
+              firstChar=$(echo "${answer^^}" | cut -c1-1)
+              if [[ "$firstChar" == "N" || "$firstChar" == "Y" ]]; then
+                renameDocker=$firstChar
+                break
               fi
-            fi
-
-            # backup all existing .nr.conf files
-            touch {{ .TMP_DIR}}/existing_nr_configs
-            local EXISTING_NR_CONFIG_FILES=$(find "$CONFIG_ROOT" -type f -name "*.nr.conf" >> {{ .TMP_DIR }}/existing_nr_configs)
-            for EXISTING_CONFIG in $(cat {{ .TMP_DIR }}/existing_nr_configs)
-            do
-              cp $EXISTING_CONFIG $EXISTING_CONFIG.bkp
-              rm $EXISTING_CONFIG
+              echo -e "Please type Y or N only."
             done
+          fi
 
-            # then re-create with current appname
-            local CONFIG_FILE="$CONFIG_ROOT/$APP_NAME.nr.conf"
-            echo "Creating $CONFIG_FILE"
-            touch $CONFIG_FILE
-            chown $USER $CONFIG_FILE
-            chmod 711 $CONFIG_FILE
-
-            # rewrite CATALINA_OPTS
-            echo "# ---- New Relic switch automatically added on $(date)" >> $CONFIG_FILE
-            echo "CATALINA_OPTS=\"\$CATALINA_OPTS -javaagent:/opt/newrelic/$APP_NAME/newrelic.jar\"" >> $CONFIG_FILE
-
-            save_newrelic_configuration $SERVICE $USER $APP_NAME
-
-            touch {{.TMP_DIR}}/tomcat_configured
+          tee -a /etc/newrelic-java/dynamic-attach.sh > /dev/null <<"EOT"
+          #!/bin/bash
+          build_stdout_json() {
+            message=$1
+            cat << EOF
+            {
+               "name": "java-dynamic-attach",
+               "protocol_version": "3",
+               "integration_version": "1.0.0",
+               "metrics": [
+                 {
+                   "event_type": "JavaDynamicAttachSample",
+                   "provider": "com.newrelic.virtuoso",
+                   "provider.executableName": "$message"
+                 }
+               ]
+            }
+          EOF
           }
+          license={{.NEW_RELIC_LICENSE_KEY}}
+          region={{.NEW_RELIC_REGION}}
+          appName=DEFAULT_APP
+          renameDocker=RENAME_DOCKER
+          pattern='org.apache.catalina.startup.Bootstrap'
+
+          isJavaLsiRunning=$(sudo ps aux | grep nri-lsi-java | grep -v grep | wc -l)
```

Comment:
@mbruzina we might also want to have a check to ensure the nri-lsi-java is still installed, just in case someone uninstalled that package.
If it's uninstalled, we should also exit without doing anything.